bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/AbstractContainerImpl.java

```diff
@@ -196,7 +196,11 @@
     public String getBackgroundStyle() {
         if (this.backgroundStyle == null) {
             StringBuilder styleBuilder = new StringBuilder();
-            getBackgroundImage().ifPresent(image -> styleBuilder.append("background-image:url(").append(image).append(");background-size:cover;background-repeat:no-repeat;"));
+            getBackgroundImage().ifPresent(image -> {
+                styleBuilder.append("background-image:url(")
+                    .append(image.replace(" ","%20"))
```

Comment:
why not using `java.net.URI` (as outlined in https://stackoverflow.com/a/65805013) for doing the encoding. That would also take care of other characters not valid in a URI.

Reply:
resourceResolver.map() would probably even better as that takes into account both mapping and does URL escaping.

Reply:
@kwin both work so thank you for your input :)

Reply:
.. and @vladbailescu even suggested a 3rd one :) let me check this one as well.

Reply:
Proposed a new approach in https://github.com/adobe/aem-core-wcm-components/pull/1493